pandas/_libs/missing.pyx

```diff
@@ -278,3 +278,126 @@ cdef inline bint is_null_period(v):
     # determine if we have a null for a Period (or integer versions),
     # excluding np.datetime64('nat') and np.timedelta64('nat')
     return checknull_with_nat(v)
+
+
+# -----------------------------------------------------------------------------
+# Implementation of NA singleton
+
+
+def _create_binary_propagating_op(name):
+    import numbers
```

Comment:
import at top?

Reply:
> import at top?

This is a python import, so was thinking to keep it here for that reason? (the top imports are only cimports)

That said, there might be a cython / cpython API to do the same?

Reply:
this should be at the top, likely cython just *works* correct with the numbers module (and makes it an optimized call), even w/o cimport.

Reply:
i haven't found a cimport-able version of `numbers`, but that would be really nice.  non-pandas pure-python imports can go at the top.  This gets executed at import-time anyway.

Reply:
OK, moved to the top